examples/server/server.cpp

```diff
@@ -479,58 +731,47 @@ struct llama_server_context
         // or the first element of the json_prompt array is a string.
         std::vector<llama_token> prompt_tokens;
 
-        if (json_prompt.is_array())
-        {
+        if (json_prompt.is_array()) {
             bool first = true;
-            for (const auto& p : json_prompt)
-            {
-                if (p.is_string())
-                {
+            for (const auto & p : json_prompt) {
+                if (p.is_string()) {
                     auto s = p.template get<std::string>();
+
                     std::vector<llama_token> p;
-                    if (first)
-                    {
+                    if (first) {
                         p = ::llama_tokenize(ctx, s, add_bos, TMP_FORCE_SPECIAL);
                         first = false;
-                    }
-                    else
-                    {
+                    } else {
                         p = ::llama_tokenize(ctx, s, false, TMP_FORCE_SPECIAL);
                     }
+
                     prompt_tokens.insert(prompt_tokens.end(), p.begin(), p.end());
-                }
-                else
-                {
-                    if (first)
-                    {
+                } else {
+                    if (first) {
                         first = false;
                     }
+
                     prompt_tokens.push_back(p.template get<llama_token>());
                 }
             }
-        }
-        else
-        {
+        } else {
             auto s = json_prompt.template get<std::string>();
             prompt_tokens = ::llama_tokenize(ctx, s, add_bos, TMP_FORCE_SPECIAL);
         }
 
         return prompt_tokens;
     }
 
-    server_slot* get_slot(int id) {
+    server_slot * get_slot(int id) {
         int64_t t_last = ggml_time_us();
-        server_slot *last_used = nullptr;
+        server_slot * last_used = nullptr;
 
-        for (server_slot & slot : slots)
-        {
-            if (slot.id == id && slot.available())
-            {
+        for (server_slot & slot : slots) {
+            if (slot.id == id && slot.available()) {
                 return &slot;
             }
 
-            if (slot.available() && slot.t_last_used < t_last)
-            {
+            if (slot.available() && slot.t_last_used < t_last) {
```

Comment:
I'm having trouble understanding the idea of keeping track of `t_last_used` here. If someone know, can you explain it? Thank you!

Reply:
Oh I see, I misunderstood and thought that we're taking the **last** used slot (while we actually take the **least** used slot). That makes sense now, thank you.